Sources/FoundationEssentials/FileManager/FileManager+Files.swift

```diff
@@ -362,32 +378,86 @@ extension _FileManagerImpl {
         isDirectory = result.isDirectory
         return true
     }
-    
+
+#if !os(Windows)
     private func _fileAccessibleForMode(_ path: String, _ mode: Int32) -> Bool {
         path.withFileSystemRepresentation { ptr in
             guard let ptr else { return false }
             return access(ptr, mode) == 0
         }
     }
-    
+#endif
+
     func isReadableFile(atPath path: String) -> Bool {
+#if os(Windows)
+        do {
+            return try path.withNTPathRepresentation {
+                var faAttributes: WIN32_FILE_ATTRIBUTE_DATA = .init()
+                return GetFileAttributesExW($0, GetFileExInfoStandard, &faAttributes) == true
+            }
+        } catch {
+            return false
+        }
+#else
         _fileAccessibleForMode(path, R_OK)
+#endif
     }
     
     func isWritableFile(atPath path: String) -> Bool {
+#if os(Windows)
+        do {
+            return try path.withNTPathRepresentation {
```

Comment:
nit: do you think it's better to write it like this or maybe as `try? path.withNTPathRepresentation { ... } ?? false` if you want, just to avoid some extra nesting - thoughts?

Reply:
As long as I can get it to compile, I think I prefer the `try? ... ?? false` variant.